tests/unit/backend/data/test_LocalDataService.py

```diff
@@ -128,321 +161,124 @@ def getMockInstrumentConfig():
+    # test that `readStateConfig` reads the `GroupingMap` from its separate JSON file.
+    groupingMapPath = Resource.getPath("inputs/pixel_grouping/groupingMap.json")
+    parametersPath = Resource.getPath("inputs/calibration/CalibrationParameters.json")
+    expected = Calibration.parse_file(parametersPath)
+    localDataService = LocalDataService()
+    with state_root_redirect(localDataService, stateId=expected.instrumentState.id.hex) as tmpRoot:
+        indexor = localDataService.calibrationIndexor("57514", True)
+        tmpRoot.addFileAs(parametersPath, indexor.parametersPath(1))
+        indexor.reconcileIndexToFiles()  # NOTE manually update the Indexor
+        assert not localDataService._groupingMapPath(tmpRoot.stateId).exists()
+        localDataService._prepareStateRoot = mock.Mock(
+            side_effect=lambda x: tmpRoot.addFileAs(groupingMapPath, localDataService._groupingMapPath(x))
         )
-
-        with tempfile.TemporaryDirectory(prefix=Resource.getPath("outputs/")) as tmpDir:
-            stateId = "ab8704b0bc2a2342"
-            stateRootPath = Path(tmpDir) / stateId
-            os.makedirs(stateRootPath)
-
-            defaultGroupingMapFilePath = Resource.getPath("inputs/pixel_grouping/defaultGroupingMap.json")
-            groupingMapFilePath = stateRootPath / "groupingMap.json"
-            localDataService._groupingMapPath = mock.Mock(return_value=groupingMapFilePath)
-
-            # Write a 'groupingMap.json' file to the <state root>, with the _correct_ stateId;
-            #   note that the _value_ of the stateId field is _not_ validated at this stage, except for its format.
-            with open(defaultGroupingMapFilePath, "r") as f:
-                groupingMap = GroupingMap.model_validate_json(f.read())
-            groupingMap.coerceStateId(ObjectSHA(hex=stateId, decodedKey=None))
-            localDataService._writeGroupingMap(stateId, groupingMap)
-
-            # read it back
-            with open(groupingMapFilePath, "r") as f:
-                groupingMap = GroupingMap.model_validate_json(f.read())
-            with open(defaultGroupingMapFilePath, "r") as f:
-                defaultGroupingMap = GroupingMap.model_validate_json(f.read())
-
-        # test that relative paths are preserved
-        relativePathCount = 0
-        for n, focusGroup in enumerate(groupingMap.liteFocusGroups):
-            assert focusGroup == defaultGroupingMap.liteFocusGroups[n]
-            if not Path(focusGroup.definition).is_absolute():
-                relativePathCount += 1
-        for n, focusGroup in enumerate(groupingMap.nativeFocusGroups):
-            assert focusGroup == defaultGroupingMap.nativeFocusGroups[n]
-            if not Path(focusGroup.definition).is_absolute():
-                relativePathCount += 1
-        assert relativePathCount > 0
-    finally:
-        Config._config["instrument"]["calibration"]["powder"]["grouping"]["home"] = savePath
```

Comment:
All of these were moved to near the end of the file, to correspond to the location of the methods they test.